backend/data_formatter.py

```diff
@@ -0,0 +1,71 @@
+import pandas as pd
+
+class DataFormatter:
+    def __init__(self, df_to_format: pd.DataFrame):
+        self._df = df_to_format.copy()
+
+
+    def get_formatted_df(self):
+        return self._df
+
+
+    def unbias(self):
+        self._df =  self._df.apply(DataFormatter._unbias_row, axis=1)
+        return self
+
+
+    @staticmethod
+    def _unbias_row(row):
+        # TODO only flip bits that have bias == 1
+        if row['confusion_value'] == 'FP' and row['Bias'] == 1:
+            row['confusion_value'] = 'TN'
+            row['Bias'] = 0
+        return row
+
+
+    def filter_by(self, filter_gender: str = None, filter_race: str = None, filter_state: str = None):
+        # ---- TODO: remove this code once PR with the updated filtering code is merged
+        if filter_gender in ['Female', 'Male', 'Non-Binary', 'Other']:  # Check for Gender
+            self._df = self._df[self._df['Gender'] == filter_gender]
+        # ----
+        # if filter_gender:
+        #     # call formatter_gender on self.formatted_df
+        # if filter_race:
+        #     # call formatter_race on self.formatted_df
+        # if filter_state:
+        #     # call formatter_state on self.formatted_df
+        return self
+
+
+    def filter_invalid_transactions(self):
+        """Creates a new dataset without any rows marked as blocked (i.e. False Positive (incorrectly blocked), True Positive (correctly blocked))
+        """
+        self._df = self._df[(self._df['confusion_value'] != 'FP') & (self._df['confusion_value'] != 'TP')]
+        return self
+
+
+    def _clean_data(self):
```

Comment:
I see that you copied the old codes here, which should work.
But you might wanna change the doc here to "remove columns", because I'm removing those columns that are empty. The reason I did remove columns is because with Julie's initial dataset, all rows have missing fields: if I removed those rows, there will be nothing left.